src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/PropertiesManager.java

```diff
@@ -108,12 +108,10 @@ public MicroProfileProjectInfo getMicroProfileProjectInfo(Module module,
         if (module != null) {
             SearchScope scope = createSearchScope(module, scopes, classpathKind == ClasspathKind.TEST);
             SearchContext context = new SearchContext(module, scope, collector, utils, documentFormat);
-            DumbService.getInstance(module.getProject()).runReadActionInSmartMode(() -> {
-                Query<PsiModifierListOwner> query = createSearchQuery(context);
-                beginSearch(context, monitor);
-                query.forEach((Consumer<? super PsiModifierListOwner>) psiMember -> collectProperties(psiMember, context, monitor));
-                endSearch(context, monitor);
-            });
+            Query<PsiModifierListOwner> query = createSearchQuery(context);
+            beginSearch(context, monitor);
+            query.forEach((Consumer<? super PsiModifierListOwner>) psiMember -> collectProperties(psiMember, context, monitor));
+            endSearch(context, monitor);
```

Comment:
if an exception can be thrown during the search, then you prolly want to execute `endSearch` in a try/finally block

Reply:
fixed